habitat_sim/agent/agent.py

```diff
@@ -99,12 +100,18 @@ class Agent(object):
         graph in python is dangerous due to differences in c++ and python memory management
     """
 
+    # TODO: is there a way to convince attrs to construct `body` from a
+    # `SceneNode` argument in the constructor? This works around that by saving
+    # scene node into an attribute and then passing that to AttachedObject
+    # constructor in a post-init. UGLY.
```

Comment:
I am not aware of one, but you can add `init=False` to the `@attr.s(...)` (https://www.attrs.org/en/stable/api.html#attr.s) and then implement `__init__` method yourself!

Reply:
I tried a few attempts at providing my own `__init__`, but failed miserably -- though I only tried to add `init=False` to the `body` `attrib`, not to the others. I never used attrs before, but if I understand correctly, when I have my own `__init__`, I need to manually init all the attributes, right?

Reply:
There are two places you can provide `init=False`.  The first is on the attributes themselves,

```python
@attr.s(auto_attribs=True)
class MyClass:
    my_attrib: int = attr.ib(init=False, default=5)


print(MyClass())
# MyClass(my_attrib=5)

MyClass(6)
# Traceback (most recent call last):
#   File "tmp.py", line 11, in <module>
#     MyClass(6)
# TypeError: __init__() takes 1 positional argument but 2 were given
```
In this case, attrs will still write `__init__` and initialize `my_attrib`, but `my_attrib` won't be an argument in `__init__`

The second is on the decorator
```python
@attr.s(init=False, auto_attribs=True)
class MyClass:
    my_attrib: int

    def __init__(self, my_attrib_value_getter):
        self.my_attrib = my_attrib_value_getter()


print(MyClass(lambda: 12))
# MyClass(my_attrib=12)
```
In this case, `attrs` will not write an `__init__` method and so you can do whatever you want it :).  It also won't initialize or create any attributes (even if you provide them a default value, i.e. if I didn't set `self.my_attrib` above, you'd get an `AttributeError: 'MyClass' object has no attribute 'my_attrib'` if you tried to access it), but you'll still get all the other niceties of attrs like print methods!

Reply:
Thank you, works great!